### PR TITLE
Add source location to regular output

### LINKED ISSUE #918: New output format is missing line numbers
I tried this code:

```rust
fn main() {
    let mut x = 0;
    assert!(x != 1);
    x += 1;
    assert!(x != 1);
}
```

using the following command line invocation:

```
kani test.rs
```

with Kani version: `b20f72b0ca8`

The output doesn't include the line numbers of the asserts:
```
RESULTS:
Check 1: main.assertion.2
         - Status: SUCCESS
         - Description: "assertion failed: x != 1"

Check 2: main.assertion.4
         - Status: SUCCESS
         - Description: "attempt to add with overflow"

Check 3: main.assertion.6
         - Status: FAILURE
         - Description: "assertion failed: x != 1"
```
so it's not clear which of the two identical asserts failed. Whereas with the old format, the line numbers were included:
```
[main.assertion.1] line 3 assertion failed: x != 1: SUCCESS
[main.assertion.2] line 4 attempt to add with overflow: SUCCESS
[main.assertion.3] line 5 assertion failed: x != 1: FAILURE
```

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

Parse CBMC property sourceLocation field and print the location of the assertion that has failed. Output looks like:

```
Check 21: main.undefined-shift.1
         - Status: FAILURE                          
         - Description: "shift distance too large"                                                                                                                                 
         - Location: /home/ubuntu/kani/tests/ui/check_operations/operations.rs:17:13 at function main
...

Check 75: std::ptr::const_ptr::<impl *const T>::add.overflow.1
         - Status: SUCCESS
         - Description: "arithmetic overflow on unsigned to signed type conversion"
         - Location: /rustc/1eb72580d076935a3e590deb6e5813a5aef3eca4/library/core/src/ptr/const_ptr.rs:576:30 at function std::ptr::const_ptr::<impl *const T>::add

Check 76: pointer_arithmetic.13
         - Status: SUCCESS
         - Description: "pointer arithmetic: pointer NULL"

```

### Resolved issues:

Fix #918

### Call-outs:

Some CBMC checks seem to be missing location. In those cases, we do not print the `Location` line. The std library path is somewhat broken, since it does not include the rustc path. But this is unrelated to this change.

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by tautschnig** [comment]
CBMC 5.53.0 has been marked for release (https://github.com/diffblue/cbmc/pull/6734), but the actual release still seems to be missing. I've asked Diffblue about it.

**Comment by tautschnig** [comment]
The actual 5.53.0 release of CBMC is now available.

**Comment by zhassan-aws** [comment]
The CBMC version was bumped to 5.53.0 in #965. CBMC's json output now includes the source location, e.g.
```json
      {
        "description": "[KANI_CHECK_ID_iss918.91812645::iss918_0] assertion failed: x != 1",
        "property": "main.assertion.2",
        "sourceLocation": {
          "column": "5",
          "file": "/home/ubuntu/examples/iss918/iss918.rs",
          "function": "main",
          "line": "3"
        },
        "status": "SUCCESS"
      },
```

File: scripts/cbmc_json_parser.py [hunk 1]
```diff
@@ -72,6 +76,42 @@ def main():
...
+           {
+             'column': '<num>',
+             'file': '<file_name>',
+             'function': '<fn_name>',
+             'line': '<num>'
+           }},
+
+           Some fields might be missing.
+        """
+        self.fname = source_location.get("file", None)
```
Comment (tedinski) [inline scripts/cbmc_json_parser.py:93]: nit: fname isn't great, I thought it was function name until I saw the next line. Just use filename?

**Review by tedinski — APPROVED** [review]

File: scripts/cbmc_json_parser.py (hunk 1 above)
Comment (celinval) [reply scripts/cbmc_json_parser.py:93]: For sure!

File: scripts/cbmc_json_parser.py [hunk 2]
```diff
@@ -72,6 +76,42 @@ def main():
...
+        if self.filename:
+            s = f"{self.filename}"
+            if self.line:
+                s += f":{self.line}"
+                if self.column:
+                    s += f":{self.column}"
+        else:
+            s = "Unknown File"
+        if self.function:
+            s += f" at function {self.function}"
```
Comment (zhassan-aws) [inline scripts/cbmc_json_parser.py:108]: I think "in function" is more appropriate than "at function".

**Review by zhassan-aws — APPROVED** [review]

File: scripts/cbmc_json_parser.py (hunk 2 above)
Comment (celinval) [reply scripts/cbmc_json_parser.py:108]: Sure

Comment (celinval) [inline scripts/cbmc_json_parser.py:108]:
```suggestion
            s += f" in function {self.function}"
```

File: scripts/cbmc_json_parser.py [hunk 3]
```diff
@@ -570,9 +609,12 @@ def construct_property_message(properties):
             Property 53: sinf.assertion.1
                 - Status: SUCCESS
                 - Description: "assertion false"
+                - Location: file/path.rs:10:8 at function harness
```
Comment (celinval) [inline scripts/cbmc_json_parser.py:612]:
```suggestion
                - Location: file/path.rs:10:8 in function harness
```

File: tests/ui/regular-output-format-pass/expected [hunk 4]
```diff
@@ -1 +1,3 @@
-Description: "assertion failed: 1 + 1 == 2"
+Description: "assertion failed: 1 + 1 == 2"\
+Location:
+tests/ui/regular-output-format-pass/main.rs:7:5 at function main
```
Comment (celinval) [inline tests/ui/regular-output-format-pass/expected:3]:
```suggestion
tests/ui/regular-output-format-pass/main.rs:7:5 in function main
```

**Pull request merged**